packages/react/src/ReactHooks.js

```diff
@@ -234,8 +234,10 @@ export function useMemoCache(size: number): Array<any> {
   return dispatcher.useMemoCache(size);
 }
 
-export function useEffectEvent<T>(callback: T): void {
+export function useEffectEvent<Args, Return, F: (...Array<Args>) => Return>(
```

Comment:
@kassens Do we actually need the `Args` and `Return` type parameter here? In TS we can just do `useEffectEvent<T extends Function>`. Does something similar exist in Flow?